pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -274,3 +278,36 @@ func isCustomizationPendingError(err error) bool {
 	}
 	return false
 }
+
+// linuxGuestIDPrefixes is derived from the vimTypes.VirtualMachineGuestOsIdentifier values.
+var linuxGuestIDPrefixes = [...]string{
```

Comment:
Does it make sense to add this list and the `isLinuxGuest` methods to govmomi  and reference from there?

Govmomi already seems to have enums for all the guest OSes:

https://github.com/vmware/govmomi/blob/main/vim25/types/enum.go#L8316

Reply:
It would nice but we'd want to come along from the auto generated but it looks like that info isn't provided along with those enums.

You can tease the OSFamily from the GuestID out of the EnvironmentBrowser but the ergonomics to do at this point in the code is just not that friendly and just don't think it is worth it to do so for right now.